### PR TITLE
完成任务：对应的功能描述

### PR DESCRIPTION
**Pull request by wzy205627**
# Summary

# Related issues

# Changes these areas

- [ ] Bugfixes
- [ ] Feature behavior
- [ ] Command line interface
- [ ] Configuration options
- [ ] Internal architecture
- [ ] Snapshot data layout on disk

---
## Summary by cubic
Introduce strict URL validation and unified error handling across API, CLI, and web UI. Invalid inputs are blocked early, and 400/403/404/500 now return clear JSON for `/api/*` and a new HTML page for the UI.

- **New Features**
  - Added validators in `archivebox.misc.util`: `validate_url_strict`, `validate_urls_list`, `contains_invisible_chars`, and `MAX_URL_LENGTH`.
  - Enforced URL checks in API `create_snapshot`, CLI `archivebox add`, and `core.forms` (per-URL validation).
  - Added `GlobalExceptionMiddleware` to surface invalid URL/validation errors as 400s; enabled in settings.
  - Implemented 400/403/404/500 handlers in `core.urls` with JSON for API and `templates/core/400.html` for web.
  - New 400 HTML template with simple guidance for fixing URLs.
  - Tests added for new validators and edge cases in `tests/test_util.py`.

<sup>Written for commit bb2d7b5d1663ca37bb0c4ff4b9cf5f9a01f202e7. Summary will update on new commits.</sup>

### CODE REVIEW

File: archivebox/core/exception_middleware.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        if hasattr(exception, 'messages'):
+            for msg in exception.messages:
+                urls = list(find_all_urls(str(msg)))
+                if urls:
+                    return urls[0]
+    
+    return None
+
+
+class GlobalExceptionMiddleware:
```
Comment (pirate) [inline archivebox/core/exception_middleware.py:54]: I think it would be better to set up werkzeug or something than implement this manually.

**Comment by pirate** [comment]
Whats wrong with the django defaults and our existing URL validation? please open an issue first next time to discuss the problem you're trying to solve
